webpack/components/Table/EmptyStateMessage.js

```diff
@@ -81,9 +82,18 @@ const EmptyStateMessage = ({
           {emptyStateBody}
         </EmptyStateBody>
         {showPrimaryAction &&
-          <Button>
+          <Button
+            ouiaId={emptyStateButtonDetails?.ouiaId}
+            onClick={showEmptyStateButtonClickAction ?
+              () => emptyStateButtonClickActions() : undefined}
+            variant={emptyStateButtonDetails?.variant}
+            aria-label={emptyStateButtonDetails?.aria_label}
+          >
+            {showEmptyStateButtonClickAction ? primaryActionTitle :
```

Comment:
What about passing the entire button as a prop?  This would allow us to eliminate several props - `showEmptyStateButtonClickAction`, `emptyStateButtonClickActions`, and `emptyStateButtonDetails`.

Instead, you'd pass a single `primaryActionButton` prop similar to this:
```js
primaryActionButton={<Button
                                         ouiaId={...}
                                         onClick={...}
                                         variant={...}
                                         aria-label={...}
>...</Button>

```

You could make this a nonrequired prop with a default of `undefined`.  Then here, in the component, you could say
```js
const actionButton = primaryActionButton ?? (
<Button>
  <a href={primaryActionLink} style={{ color: 'white', textDecoration: 'none' }}>{primaryActionTitle}</a>
            }
</Button>
);
```

and then here, where it's rendered,
```js
{showPrimaryAction ? primaryActionButton : null}
```

Thoughts?

Reply:
oh, and I don't think this change should affect the tests :wink:

Reply:
@jeremylenz I'm a bit confused on why you said {showPrimaryAction ? primaryActionButton : null}? Would not that be {showPrimaryAction ? actionButton : null} in your code example?

Reply:
@tazhibaevaaliya my bad!  You're right. I will edit it.

Reply:
Should it include return?